app/controllers/MessagesController.scala

```diff
@@ -129,7 +128,7 @@ class MessagesController @Inject()(
         implicit request =>
           val result = for {
             departure <- OptionT(departureRepository.getWithoutMessages(departureId, request.channel))
-            if departure.eoriNumber == request.eoriNumber
+            if request.hasMatchingEori(departure)
```

Comment:
Should this be hasMatchingEnrolmentId ?

Reply:
Yes you're probably right

Reply:
Ok should be addressed in the latest commit @crystalmark